src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -253,27 +254,96 @@ ExpressionEvaluator::_evaluate_in_expression<ExpressionEvaluator::Bool>(const In
   if (right_expression.type == ExpressionType::List) {
     const auto& array_expression = static_cast<const ListExpression&>(right_expression);
 
+    if (left_expression.data_type() == DataType::Null) {
+      // `NULL IN ...` is NULL
+      return std::make_shared<ExpressionResult<ExpressionEvaluator::Bool>>(std::vector<ExpressionEvaluator::Bool>{0},
+                                                                           std::vector<bool>{1});
+    }
+
+    if (left_expression.data_type() == DataType::Null || array_expression.elements().empty()) {
```

Comment:
`left_expression.data_type() == DataType::Null` is already handled in the if above - and wouldn't be False anyway, at least not in sqlite

Reply:
copy paste